test/clj/rems/test/api/applications.clj

```diff
@@ -813,6 +814,73 @@
           "user should be forbidden to send command"))
 
     (testing "send commands with authorized user"
+      (testing "even handler cannot comment without request"
+        (is (= {:errors [{:type "forbidden"}], :success false}
+               (send-dynamic-command handler-id
+                                     {:type :rems.workflow.dynamic/comment
+                                      :application-id application-id
+                                      :comment "What am I commenting on?"}))))
+      (testing "comment with request"
+        (let [eventcount (count (get-in (get-application handler-id application-id)
+                                        [:application :dynamic-events]))]
+          (testing "commenter can't comment before request"
+            (is (= {:errors [{:type "forbidden"}], :success false}
+                   (send-dynamic-command commenter-id
+                                         {:type :rems.workflow.dynamic/comment
+                                          :application-id application-id
+                                          :comment "Not yet?"}))))
+          (testing "requesting comment"
+            (is (= {:success true} (send-dynamic-command handler-id
+                                                         {:type :rems.workflow.dynamic/request-comment
+                                                          :application-id application-id
+                                                          :commenters [decider-id commenter-id]
+                                                          :comment "What say you?"})))
+            (is (= #{commenter-id decider-id}
+                   (set (get-in (get-application handler-id application-id)
+                               [:application :commenters])))))
+          (testing "cannot see internal commenter state"
+            (is (= nil (get-in (get-application handler-id application-id)
+                               [:application :latest-comment-request-by-user]))))
```

Comment:
Removing `:latest-comment-request-by-user` from the schema will avoid the need for this test.